### DISCUSSION TITLE
GitHub Copilot

### DISCUSSION THREAD

**Discussion by muhammad-arefin-23** [community/community#67998] [Copilot Conversations]
### Select Topic Area

Question

### Body

I'm currently working on a project where we're using Git for version control. I've come across a scenario where I need to merge two branches, but I want to keep all the commit history from both branches intact. Can anyone recommend the best Git workflow or commands to achieve this while preserving the commit history?

**Comment by Md-arefin** [comment]
To merge two branches while preserving commit history in Git, use the `git merge --no-ff <source_branch>` command, where `<source_branch>` is the branch you want to merge from. The `--no-ff` flag ensures that a merge commit is created, maintaining the history of both branches. This approach helps track individual branch histories and provides clarity when reviewing changes.

**Comment by Md-arefin** [comment]
To merge two branches while preserving the commit history from both branches in Git, use the `git merge --no-ff <source_branch>` command. This command ensures that a merge commit is created, capturing the full history of changes from both branches, even if Git could perform a fast-forward merge.

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
